### ISSUE TITLE
SDL_JoystickGetGUIDString crashes CPython interperter but not PyPy

### ISSUE THREAD

**Issue by marcusva** [marcusva/py-sdl2#75]
Originally reported by: **Arthur Yidi (Bitbucket: [ayidi](https://bitbucket.org/ayidi), GitHub: Unknown)**

----------------------------------------

```
#!python

SDL_Init(SDL_INIT_VIDEO | SDL_INIT_JOYSTICK | SDL_INIT_GAMECONTROLLER)
numJoys = SDL_NumJoysticks()
joyA = SDL_JoystickOpen(0)
SDL_JoystickNameForIndex(0)
SDL_JoystickGetAttached(joyA)
SDL_JoystickInstanceID(joyA)
guidA = SDL_JoystickGetGUID(joyA)

buff = create_string_buffer(33)
size = sizeof(buff)
# crash here
SDL_JoystickGetGUIDString(guidA, buff, size)
```

----------------------------------------
- Bitbucket: https://bitbucket.org/marcusva/py-sdl2/issue/75

**Comment by Kentzo** [comment]
@marcusva Would you accept a PR with that fix?

**Comment by marcusva** [comment]
@Kentzo: of course.

**Comment by rLinks234** [comment]
I am also seeing this issue. In addition to the mentioned function, `SDL_GameControllerMappingForGUID` also causes a SIGABRT on my Linux x86_64 machine (Fedora 27). Coincidentally, it also requires a guid struct (non-pointer) as a parameter

**Comment by a-hurst** [comment]
@rLinks234 That function also calls SDL_JoystickGetGUIDString internally, so that might also be the cause of the issue.

It looks like the underlying Python bug was fixed in both 3.8 and the latest version of 3.7, but that still means the function is broken in 2.7 and 3.5/3.6 (and earlier versions of 3.7). Since it'd be nice for things not to be broken on older versions, I think it should be possible to have pure-Python implementations of both. I'll work on this and see if I get a decent solution in place for 0.9.7.
